File: Model/DTO/RecipeDto.cs
using EFGetStarted.Attributes;

namespace EFGetStarted.Model.DTO;

public class RecipePostDto : AbstractPostDto
{
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<int> IngredientGroupIds { get; set; }
    public long PreparationDuration { get; set; } = 0;
    public long CookingDuration { get; set; }
}

public class RecipeFullPostDto : AbstractPostDto
{
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    
    public List<IngredientGroupFullPostDto> IngredientGroups { get; set; }
    public long PreparationDuration { get; set; } = 0;
    public long CookingDuration { get; set; }
}

public class RecipePutDto : AbstractPutDto
{
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<int> IngredientGroupsIds { get; set; }
    public long PreparationDuration { get; set; } = 0;
    public long CookingDuration { get; set; }
}

public class RecipeFullPutDto : AbstractPutDto
{
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<IngredientGroupFullPutDto> IngredientGroups { get; set; }
    public long PreparationDuration { get; set; } = 0;
    public long CookingDuration { get; set; }
}
public class RecipeGetDto : AbstractGetDto
{
    [RecipeCodeValidation]
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<IngredientGroupGetDto> IngredientGroups { get; set; }
    public long PreparationDuration { get; set; }
    public long CookingDuration { get; set; }

    public long TotalDuration => PreparationDuration + CookingDuration;
    
    public bool IsFavorite { get; set; }
}

File: src/main/kotlin/com/example/recipecollection/dto/RecipeDto.kt
package com.example.recipecollection.dto

import jakarta.validation.constraints.NotBlank

data class RecipeDto(
    val id: Long?,
    val code: String,
    val name: String,
    val description: String,
    val preparationDuration: Long,
    val cookingDuration: Long,
    val totalDuration: Long,
    val createdByUserId: Long,
    val imgBase64: String?,
    val ingredientGroups: List<IngredientGroupDto>,
    val isFavorite: Boolean = false,
)

data class RecipeRequest(
    @field:NotBlank val code: String,
    @field:NotBlank val name: String,
    @field:NotBlank val description: String,
    val preparationDuration: Long = 0,
    val cookingDuration: Long,
    val createdByUserId: Long,
    val imgBase64: String? = null,
    val ingredientGroupIds: List<Long> = emptyList(),
)
